asn1/LedgerLedgerLotocol.asn

```diff
@@ -0,0 +1,135 @@
+LedgerLedgerLotocol
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt128,
+    UInt256
+    FROM GenericTypes
+
+    Address,
+    Timestamp
+    FROM InterledgerTypes
+;
+
+FreeRequest ::= SEQUENCE {
+    -- Used to associate requests and corresponding responses
+    -- MUST be either zero and non-repeating
+    -- If requestId = 0, the server MUST not send a response
+    requestId UInt128,
+    packet InterledgerPacket
+}
+
+FreeResponse ::= SEQUENCE {
+    -- Used to associate response to earlier FreeRequest
+    requestIdRef UInt128,
+    packet InterledgerPacket
+}
+
+PaidRequest ::= SEQUENCE {
```

Comment:
The `PaidRequest` is an unconditional transfer. It wouldn't make that much sense to put a protocol other than an Interledger Payment Packet on a conditional transfer. You would need to figure out another way for the recipient of it to know the fulfillment, which probably wouldn't provide a lot of value.